pkg/sitewise/datasource.go

```diff
@@ -41,15 +43,35 @@ func NewDatasource(settings backend.DataSourceInstanceSettings) (*Datasource, er
 			Settings: cfg,
 		}
 
-		authInfo, err := edgeAuthenticator.Authorize()
+		var waitTime time.Duration
+
+		updateAuth := func() error {
+			authInfo, err := edgeAuthenticator.Authorize()
+			if err == nil {
+				cfg.AccessKey = authInfo.AccessKeyId
+				cfg.SecretKey = authInfo.SecretAccessKey
+				cfg.SessionToken = authInfo.SessionToken
+				cfg.AuthType = awsds.AuthTypeKeys
+				waitTime = time.Until(authInfo.SessionExpiryTime)
+				log.DefaultLogger.Debug("should wait for: ", "time:", waitTime)
+				waitTime = 10 * time.Second
```

Comment:
Why 10 times here.

Reply:
just trying to test something here for now - will clean this up next after some testing